### PR TITLE
Fix updateContact method according to API manual

### PR DESCRIPTION
**Pull request by webarchitect609**
Append 'createIfNotExists' option to updateContact method;

I'm not sure how to format very long `\Snowcap\Emarsys\Client::send` call in `\Snowcap\Emarsys\Client::updateContact`.

### CODE REVIEW

File: src/Snowcap/Emarsys/Client.php [hunk 1]
```diff
@@ -250,17 +250,24 @@ public function createContact(array $data)
      * Updates one or more contacts/recipients, identified by an external ID.
      *
      * @param array $data
+     * @param bool $createIfNotExists When enabled, if the contact does not exist in the database, it is created
+     * automatically.
      * @return Response
+     * @link http://documentation.emarsys.com/resource/developers/endpoints/contacts/update-contacts/
      */
-    public function updateContact(array $data)
+    public function updateContact(array $data, $createIfNotExists = false)
```
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: Let's avoid optional parameters. Implementing a new method like `updateContactAndCreateIfNotExists` would be my preferred way.

File: src/Snowcap/Emarsys/Client.php [hunk 2]
```diff
@@ -250,17 +250,24 @@ public function createContact(array $data)
      * Updates one or more contacts/recipients, identified by an external ID.
      *
      * @param array $data
+     * @param bool $createIfNotExists When enabled, if the contact does not exist in the database, it is created
+     * automatically.
      * @return Response
+     * @link http://documentation.emarsys.com/resource/developers/endpoints/contacts/update-contacts/
```
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: Please remove this link as it could be changed in future.

Comment (webarchitect609) [inline src/Snowcap/Emarsys/Client.php]: Done. 

But that was a link to official API documentation. So I thought it's nice to use just `Shift+F1` in my PhpStorm when I'm confusing, for example, what I should pass in `$data`.

File: src/Snowcap/Emarsys/Client.php (hunk 1 above)
Comment (webarchitect609) [inline src/Snowcap/Emarsys/Client.php]: OK. Will push changes just in minute.

File: src/Snowcap/Emarsys/Client.php (hunk 2 above)
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: If you would like to have it than it must be added to all methods. Possible in a separate PR.

File: src/Snowcap/Emarsys/Client.php [hunk 3]
```diff
@@ -264,6 +264,24 @@ public function updateContact(array $data)
     }
 
     /**
+     * Updates one or more contacts/recipients, identified by an external ID. If the contact does not exist in the
+     * database, it is created.
```
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: Code comments are generally evil as they start lying with a pass of time. If you think more description is required maybe a method name is not good enough.

File: src/Snowcap/Emarsys/Client.php [hunk 4]
```diff
@@ -264,6 +264,24 @@ public function updateContact(array $data)
...
+     *
+     * @param array $data
```
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: Should be `mixed[]` I guess. Does not help much but at least shows that developer thought about it. 😊

File: src/Snowcap/Emarsys/Client.php [hunk 5]
```diff
@@ -264,6 +264,24 @@ public function updateContact(array $data)
...
+     * @return Response
+     */
+    public function updateContactAndCreateIfNotExists(array $data)
+    {
+        if (isset($data['contacts']) && is_array($data['contacts'])){
+            foreach($data['contacts'] as &$contact){
+                $contact = $this->mapFieldsToIds($contact);
+            }
+        }
```
Comment (tim-bezhashvyly) [inline src/Snowcap/Emarsys/Client.php]: Now this code is duplicated. Maybe we can extract it in a utility method?

Comment (webarchitect609) [inline src/Snowcap/Emarsys/Client.php]: Done. And found one more duplication. Now 3 copies of code packed in utility method. Unfortunately, I can't test affected methods before monday :(

**Comment by webarchitect609** [comment]
Done. No mutability in `\Snowcap\Emarsys\Client::mapFieldsForMultipleContacts`

**Comment by tim-bezhashvyly** [comment]
Thank you for the commitment.

**Pull request merged**